src/libs/nav2_util/CMakeLists.txt

```diff
@@ -17,18 +17,48 @@ find_package(tf2 REQUIRED)
 
 include_directories(include)
 
-add_library(nav2_util STATIC
+add_library(costmap_lib STATIC
   src/costmap.cpp
 )
 
-ament_target_dependencies(nav2_util
+add_library(map_lib STATIC
+  src/map/map.c
+  src/map/map_store.c
+  src/map/map_range.c
+  src/map/map_draw.c
+  src/map/map_cspace.cpp
+)
+
+add_library(pf_lib STATIC
+  src/pf/pf.c
+  src/pf/pf_kdtree.c
+  src/pf/pf_pdf.c
+  src/pf/pf_vector.c
+  src/pf/eig3.c
+  src/pf/pf_draw.c
+)
+
+add_library(sensors_lib STATIC
+  src/sensors/amcl_sensor.cpp
+  src/sensors/amcl_odom.cpp
+  src/sensors/amcl_laser.cpp
+)
+
+
+ament_target_dependencies(costmap_lib
+  map_lib
```

Comment:
remove these libs from costmap_lib dependencies and add their target dependencies